state/protocol/badger/snapshot_test.go

```diff
@@ -887,13 +892,18 @@ func TestSealingSegment_FailureCases(t *testing.T) {
 
 	// SCENARIO 2b: An orphaned block is chosen as head; at this height a block other than the orphaned has been finalized.
 	t.Run("sealing segment from orphaned block", func(t *testing.T) {
+		// In this test, we create two conflicting forks. To prevent accidentally creating byzantine scenarios, where
+		// multiple blocks have the same view, we keep track of used views and ensure that each new block has a unique view.
+		usedViews := make(map[uint64]struct{})
+		usedViews[sporkRoot.View] = struct{}{}
```

Comment:
Here, we are creating conflicting forks, i.e. there is the potential for randomly generating descendants with the same view number. 

In https://github.com/onflow/flow-go/pull/6933/commits/54001f71af279d443e81f61c730efb67aeea76be I updated the test to use your methods `BlockWithParentFixtureAndUniqueView` or `BlockWithParentProtocolStateAndUniqueView` 🎉